DEVELOPMENT.md

```diff
@@ -124,6 +124,21 @@ Enable errorprone ([Error Prone Installation#IDEA](https://errorprone.info/docs/
 - In ``Java Compiler`` tab, select ``Javac with error-prone`` as the compiler,
 - Update ``Additional command line parameters`` with ``-XepExcludedPaths:.*/target/generated-(|test-)sources/.* -XepDisableAllChecks -Xep:MissingOverride:ERROR ......`` (for current recommended list of command line parameters, see the top level ``pom.xml``, the definition of the ``errorprone-compiler`` profile.
 
+### Language injection in IDE
+
+Because we work a lot with SQL, in many places you'll encounter elements annotated with the `@org.intellij.lang.annotations.Language` annotation, like in: `@Language("SQL")`. It serves two purposes:
+
+- As a documentation: when a method parameter is annotated with a language annotation, the callers are aware that the argument should be a value conforming to the specific language;
+- As an aid in static analysis: when the IDE supports it, it may use this information to highlight mis-using code, and provide things like language injection in conforming code.
+
+For this reason we recommend that you annotate all API parameters which are expecting to take a `String` containing an SQL statement (or any other language, like regular expressions). You may also annotate local variables in new code, but refrain from doing it in existing code, as it only generates noise in the commit history.
```

Comment:
Why `refrain from doing it in existing code`. IMO if we decide to annotate, we should also move towards consistency all over the codebase.

Reply:
We had an offline discussion about annotating variables and decided that there is little added value and a lot of boilerplate code, so the decision is up to the creator.
Maybe the best idea is to skip this last phrase.

Reply:
> ... but refrain from doing it in existing code ...

this is generally applicable to any refactor, nothing special here

unless you want to discourage someone from sending a PR  that just adds `@Language("SQL")` because the low-value added doesn't motivate spending review time on these? (which i could see being _possible_)

Reply:
I would remove: `but refrain from doing it in existing code, as it only generates noise in the commit history.`

Instead I would say: `You may also annotate local variables in new code, if it helps IDE to recognize the language injection`. So I would not recommend to add for case where IDE already recognized that (possibly because variable was used in the method call).